### PR TITLE
Jet Substructure and Systematics

### PR DESCRIPTION
**Pull request by btamadio**
Implements jet substructure in HelpTreeBase and fat jet systematics in TreeAlgo

### CODE REVIEW

**Comment by kratsg** [comment]
go ahead

**Comment by mattleblanc** [comment]
This would have been less work if you'd branched from my PR, but, OK -- the systematics are included here, which is nice.

File: Root/HelpTreeBase.cxx [hunk 1]
```diff
@@ -3008,40 +3011,72 @@ void HelpTreeBase::FillFatJets( const xAOD::JetContainer* fatJets ) {
       static SG::AuxElement::ConstAccessor<float> tau1_wta ("Tau1_wta");
       static SG::AuxElement::ConstAccessor<float> tau2_wta ("Tau2_wta");
       static SG::AuxElement::ConstAccessor<float> tau3_wta ("Tau3_wta");
+      static SG::AuxElement::ConstAccessor<float> tau1_WTA ("Tau1_WTA");
+      static SG::AuxElement::ConstAccessor<float> tau2_WTA ("Tau2_WTA");
+      static SG::AuxElement::ConstAccessor<float> tau3_WTA ("Tau3_WTA");
       static SG::AuxElement::ConstAccessor<float> tau21_wta ("Tau21_wta");
       static SG::AuxElement::ConstAccessor<float> tau32_wta ("Tau32_wta");
+      static SG::AuxElement::ConstAccessor<float> tau21_WTA ("Tau21_WTA");
+      static SG::AuxElement::ConstAccessor<float> tau32_WTA ("Tau32_WTA");
```
Comment (kratsg) [inline Root/HelpTreeBase.cxx]: Capitalization is off here. You have duplicates!

File: Root/HelpTreeBase.cxx [hunk 2]
```diff
@@ -3008,40 +3011,72 @@ void HelpTreeBase::FillFatJets( const xAOD::JetContainer* fatJets ) {
...
+
       if ( tau1_wta.isAvailable( *fatjet_itr ) ) {
         m_fatjet_tau1_wta.push_back( tau1_wta( *fatjet_itr ) );
+      } else if ( tau1_WTA.isAvailable( *fatjet_itr ) ){
+	m_fatjet_tau1_wta.push_back( tau1_WTA( *fatjet_itr ) );
       } else { m_fatjet_tau1_wta.push_back( -999 ); }
+
```
Comment (kratsg) [inline Root/HelpTreeBase.cxx]: again a lot of duplicates

**Comment by btamadio** [comment]
What's the failed check mean? I tested this before committing and it works fine.

**Comment by kratsg** [comment]
You'll need to wait a few minutes. The automated testing is breaking because the cvmfs cache is going haywire. I'll get this tested again soon.

**Comment by kratsg** [comment]
retest please

**Comment by kratsg** [comment]
test this please

**Comment by kratsg** [comment]
@btamadio - thanks for being patient! In the future, if you click `Details` and then `Console Output` -- you can see why something might crash.

**Pull request merged**